wcfsetup/install/files/lib/acp/page/SystemCheckPage.class.php

```diff
@@ -184,7 +219,23 @@ protected function validateMysql() {
 			}
 		}
 		
-		if ($this->results['mysql']['result'] && $this->results['mysql']['innodb']) {
+		// validate foreign keys
+		$this->results['mysql']['foreignKeys'] = true;
+		foreach ($this->foreignKeys as $table => $keys) {
+			$sql = "SHOW CREATE TABLE ". $table;
```

Comment:
Use `information_schema`.

Reply:
Just a note: Using the `INFORMATION_SCHEMA` is much, much slower then the `SHOW CREATE TABLE` solution.